### ISSUE TITLE
BlueskyRun load is called twice

### ISSUE THREAD

**Issue by gwbischof** [bluesky/databroker#492]
When accessing an entry in the catalog, BlueskyRun._load() is called twice.

**Comment by danielballan** [comment]
A minimal example to reproduce:

```py
# generate_data.py
import tempfile
from suitcase.msgpack import Serializer
from bluesky import RunEngine
from bluesky.plans import count


RE = RunEngine()

directory = tempfile.TemporaryDirectory().name
with Serializer(directory) as serializer:
    RE(count([]), serializer)


from databroker._drivers.msgpack import BlueskyMsgpackCatalog
catalog = BlueskyMsgpackCatalog(f'{directory}/*.msgpack')
```

```sh
ipython -i generate_data.py
```

Now, with prints added to `BlueskyRun.__init__` and `BlueskyRun._load`, access the data:

```py
In [1]: catalog[-1]                                                                                                                                                                           
BlueskyRun.__init__
BlueskyRun._load
Out[1]: BlueskyRun.__init__
BlueskyRun._load

Run Catalog
  uid='4c3665e5-d63f-4728-9fd2-b741363a5951'
  exit_status='success'
  2020-01-14 06:39:35.295 -- 2020-01-14 06:39:35.297
  Streams:
```

It looks like `_load` is being called once per `__init__` (good) but that `__init__` is being called twice (bad).

**Comment by danielballan** [comment]
I started to look at this in https://github.com/bluesky/databroker/pull/494 back in November. On that branch, we get the expected number of prints. But it may introduce different problems!

```py
In [1]: catalog[-1]                                                                                                                                                                           
BlueskyRun.__init__
BlueskyRun._load
Out[1]: 
Run Catalog
  uid='e4c41729-8a74-46c1-abf0-a7a7d0b65e96'
  exit_status='success'
  2020-01-14 06:51:49.890 -- 2020-01-14 06:51:49.891
  Streams:
```

**Comment by danielballan** [comment]
Closed by #494
